types/v56/types.go

```diff
@@ -2768,6 +2768,7 @@ type DiskRecordType struct {
 	BusSubType         string    `xml:"busSubType,attr,omitempty"`
 	BusTypeDesc        string    `xml:"busTypeDesc,attr,omitempty"`
 	AttachedVmCount    int32     `xml:"attachedVmCount,attr,omitempty"`
+	SharingType        string    `xml:"sharingType,attr,omitempty"`
```

Comment:
Just to confirm: is this field ignored in pre-v36.0 API versions?

Reply:
BTW, let's use the chance and remove the obsolete comments few lines above too (esp. now that we're moving away from V33.0)?

```
  // SizeB is not available in API V33.0. It is replaced by SizeMb
	//SizeB              int64   `xml:"sizeB,attr,omitempty"`
```

Reply:
not ignored:
cvc-complex-type.3.2.2: Attribute 'sharingType' is not allowed to appear in element 'Disk'.

XML is validate by XSD

Reply:
comment removed

Reply:
> not ignored:
> cvc-complex-type.3.2.2: Attribute 'sharingType' is not allowed to appear in element 'Disk'.
> 
> XML is validate by XSD

Interesting. But it is ignored by pre-V36.0 VCD versions?

Reply:
it's ignored if you fetch information, nothing will return for sharingType, but if you send on update or create you will get error which I showed.

Reply:
> but if you send on update or create you will get error which I showed.

Where did you show it?

After reading your reply, I don't understand: then how does independent disk update/create still work with previous VCD versions?

Reply:
error: cvc-complex-type.3.2.2: Attribute 'sharingType' is not allowed to appear in element 'Disk'.

If works because field sharingType is optional, old version API doesn't care about this field, new API cares if you send it

Reply:
> If works because field sharingType is optional, old version API doesn't care about this field, new API cares if you send it

And this is exactly what I wanted to know with my original question in this thread. Thanks!